### PR TITLE
Remove filterOutSchedulableSimple

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: cluster-autoscaler/core/filter_out_schedulable.go [hunk 1]
```diff
@@ -67,20 +65,11 @@ func (filterOutSchedulablePodListProcessor) Process(
 	filterOutSchedulableStart := time.Now()
 	var unschedulablePodsToHelp []*apiv1.Pod
 
-	if context.EstimatorName == estimator.BinpackingEstimatorName {
-		unschedulablePodsToHelp = filterOutSchedulableByPacking(unschedulablePods, upcomingNodes, allScheduledPods,
-			context.PredicateChecker, context.ExpendablePodsPriorityCutoff, false)
-	} else {
-		unschedulablePodsToHelp = unschedulablePods
-	}
+	unschedulablePodsToHelp = filterOutSchedulableByPacking(unschedulablePods, upcomingNodes, allScheduledPods,
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/filter_out_schedulable.go]: Add some comment here. It looks really strange that we're calling this twice like that and if I didn't already knew how it works, I imagine I'd spent a while figuring it out.

**Comment by MaciekPytel** [comment]
We should remember to add to release notes that we deleted this flag. Any suggestions on how we make sure we remember it? Maybe create draft release already and add the note there?

File: cluster-autoscaler/core/filter_out_schedulable.go (hunk 1 above)
Comment (losipiuk) [reply cluster-autoscaler/core/filter_out_schedulable.go]: I will. It gets refactored with ClusterSnapshot anyway :)

**Comment by losipiuk** [comment]
> We should remember to add to release notes that we deleted this flag. Any suggestions on how we make sure we remember it? Maybe create draft release already and add the note there?

I always go through all the commits in delta between releases when doing release notes. If everyone follows this pattern it should not slip.
The only issue is that person doing release notes may have little understanding - and not catch what should be added just based on commit message.

Possibly release notes preparation should be more collaborative effort. I have some past experiences from different projects. We can talk and discuss that offline.

**Pull request merged**